### PR TITLE
Preserve units in computed Angle

### LINKED ISSUE #16594: Computed angle should preserve its unit
Currently, Servo computes and stores computed Angle only as radian. Gecko stores computed Angle like specified Angle(radian/degree/gradian/turn). This causes some linear-gradient mochitest failures in stylo and we are losing information here.  We should fix that.

### PR DESCRIPTION
**Pull request by canova**
It was converting all angles to radians before. But other browsers preserves the angle units. Fixed that behavior.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #16594 and [Bug 1360659](https://bugzilla.mozilla.org/show_bug.cgi?id=1360659)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16674)

### CODE REVIEW

File: components/style/gecko_bindings/sugar/ns_css_value.rs [hunk 1]
```diff
@@ -173,6 +173,25 @@ impl nsCSSValue {
...
+    pub fn get_angle(&self) -> Angle {
+        unsafe {
+            Angle::from_gecko_values(self.float_unchecked(), self.mUnit)
+        }
+    }
+
+    /// Sets Angle value to this nsCSSValue.
+    pub fn set_angle(&mut self, angle: Angle) {
+        let (value, unit) = angle.to_gecko_values();
+        self.mUnit = unit;
```
Comment (emilio) [inline components/style/gecko_bindings/sugar/ns_css_value.rs]: Oh, one thing I missed, please assert here we're not leaking anything (that our unit isn't one of the `heap` ones).

**Review by emilio — CHANGES_REQUESTED** [review]

File: components/style/gecko_bindings/sugar/ns_css_value.rs (hunk 1 above)
Comment (emilio) [inline components/style/gecko_bindings/sugar/ns_css_value.rs]: (Not the unit we're setting, but the pre-existing unit)

**Comment by canaltinova** [comment]
Fixed that. That assert was your point, right?

**Comment by emilio** [comment]
How is that assert ever going to be true? We definitely expect to go from `eCSSUnit_Null` to angle, at least.

I mean to assert it's not one of the actual units that needs freeing stuff in `nsCSSValue::DoReset`, because otherwise we'd be leaking them.

File: components/style/gecko_bindings/sugar/ns_css_value.rs [hunk 2]
```diff
@@ -173,6 +173,26 @@ impl nsCSSValue {
...
+    /// or `eCSSUnit_Radian`.
+    pub fn get_angle(&self) -> Angle {
+        unsafe {
+            Angle::from_gecko_values(self.float_unchecked(), self.mUnit)
+        }
+    }
+
+    /// Sets Angle value to this nsCSSValue.
+    pub fn set_angle(&mut self, angle: Angle) {
+        debug_assert!(self.mUnit == nsCSSUnit::eCSSUnit_Null);
```
Comment (emilio) [inline components/style/gecko_bindings/sugar/ns_css_value.rs]: nit: `debug_assert_eq!`?

**Review by emilio — APPROVED** [review]
r=me, assuming it holds, thanks!

We should refactor those methods to make clear when we call them, but I think that's part of https://bugzilla.mozilla.org/show_bug.cgi?id=1361032

**Pull request merged**